webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/TemporaryStore.scala

```diff
@@ -27,6 +27,13 @@ class TemporaryStore[K, V] @Inject()(system: ActorSystem) {
       map.values.toList
     }
 
+  def findAllConditionalWithKey(predicate: K => Boolean): collection.Set[(K, V)] =
+    map.synchronized {
+      map.keySet.filter(predicate).map { key =>
```

Comment:
I don't know if there is a reason you want to return a `Set`, but if there is not 
```suggestion
      map.filterKeys(predicate)
```
should have the same result and is more concise (and hopefully more performant)